packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts

```diff
@@ -100,20 +141,72 @@ export const processProtectedData = async ({
       workerpool: underMaxPriceOrders.workerpoolorder.workerpool,
       params: {
         iexec_input_files: vInputFiles,
-        iexec_developer_logger: true,
         iexec_secrets: secretsId,
         iexec_args: vArgs,
       },
     });
 
     const requestorder = await iexec.order.signRequestorder(requestorderToSign);
 
-    const { dealid } = await iexec.order.matchOrders({
+    const { dealid, txHash } = await iexec.order.matchOrders({
       requestorder,
       ...underMaxPriceOrders,
     });
+    vOnStatusUpdate({
+      title: 'PROCESS_PROTECTED_DATA_REQUESTED',
+      isDone: true,
+      payload: {
+        txHash: txHash,
+      },
+    });
+
+    const taskId = await iexec.deal.computeTaskId(dealid, 0);
+    const taskObservable = await iexec.task.obsTask(taskId, { dealid: dealid });
+    vOnStatusUpdate({
+      title: 'CONSUME_TASK_ACTIVE',
+      isDone: true,
+      payload: {
+        taskId: taskId,
+      },
+    });
+
+    await new Promise((resolve, reject) => {
+      taskObservable.subscribe({
+        next: () => {},
+        error: (e) => {
+          vOnStatusUpdate({
+            title: 'CONSUME_TASK_ERROR',
+            isDone: true,
+            payload: {
+              taskId: taskId,
+            },
+          });
```

Comment:
Unnecessary for me.
If an error is thrown, the builder deals with it in its catch block. Sending the error in an update event like this is redundant.

Also see PR https://github.com/iExecBlockchainComputing/dataprotector-sdk/pull/276/files

Reply:
Commit: d87b9353854ec0d7169ccfe64c595f766ad00a73